torch/quantization/fx/quantize.py

```diff
@@ -266,10 +266,31 @@ def prepare(self, model, qconfig_dict, inplace=False):
     def prepare_dynamic(self, model, qconfig_dict, inplace=False):
         return self._prepare(model, qconfig_dict, inplace, is_dynamic_quant=True)
 
+    # Extract the subgraph that produces the weight for dynamically quantized node
+    # and run the subgraph to observe the weight
+    def _run_weight_observers(self, observed):
+        for node in observed.graph.nodes:
+            if node.op == 'call_function' and node.target in WEIGHT_INDEX_DICT:
+                for i, node_arg in enumerate(node.args):
+                    if i in WEIGHT_INDEX_DICT[node.target]:
```

Comment:
Should there be a separate dict for dynamic ops?

Reply:
not sure, isn't the weight index the same for static and dynamic?

Reply:
I think we should only run weight observers for dynamic quantizable ops, the dict has both static and dynamic ops.

Reply:
this pass is only called in dynamic quantization. WEIGHT_INDEX_DICT is containing floating point ops, we haven't produced static/dynamic quantized ops at this point.